doc/can-define-object.md

```diff
@@ -0,0 +1,479 @@
+ - `completed` as a property that will be type coerced into a `Boolean`
+   with an initial value of `false`.
+
+This also defines a `toggle` method that will be available on _instances_ of `Todo`.
+
+`Todo` is a constructor function.  This means _instances_ of `Todo` can be be created by
+calling `new Todo()` as follows:
+
+```js
+import { DefineObject } from "can";
+
+class Todo extends DefineObject {
+  static define = {
+    name: String,
+    completed: false
+  };
+
+  toggle() {
+    this.completed = !this.completed;
+  }
+}
+
+const myTodo = new Todo();
+myTodo.name = "Do the dishes";
+console.log( myTodo.completed ); //-> false
+
+myTodo.toggle();
+console.log( myTodo.completed ); //-> true
+```
+@codepen
+@highlight 14
+
+You can also pass initial properties and their values when initializing a `DefineObject`:
+
+```js
+import { Todo } from "//unpkg.com/can-demo-models@5";
+
+const anotherTodo = new Todo( { name: "Mow lawn", completed: true } );
+
+console.log( anotherTodo.name ); //-> "Mow lawn"
+```
+@codepen
+
+## Declarative properties
+
+Arguably `can-define-object`'s most important ability is its support of declarative properties
+that functionally derive their value from other property values.  This is done by
+defining [can-define-object.types.get getter] properties like `fullName` as follows:
+
+```js
+import { DefineObject } from "can";
+
+class Person extends DefineObject {
+  static define = {
+    first: String,
+    last: String
+  };
+
+  get fullName() {
+    return this.first + " " + this.last;
+  }
+}
+
+const person = new Person({
+	first: "Justin",
+	last: "Meyer"
+});
+
+console.log(person.fullName); //-> "Justin Meyer"
+```
+@codepen
+@highlight 9-11
+
+Now, when a `person` is created, there is a `fullName` property available like:
+
+```js
+import { Person } from "//unpkg.com/can-demo-models@5";
+
+const me = new Person( { first: "Harry", last: "Potter" } );
+console.log( me.fullName ); //-> "Harry Potter"
+```
+@codepen
+@highlight 4
+
+This property can be bound to like any other property:
+
+```js
+import { Person } from "//unpkg.com/can-demo-models@5";
+
+const me = new Person({first: "Harry", last: "Potter"});
+
+me.on( "fullName", ( ev, newValue, oldValue ) => {
+	console.log( newValue ); //-> Harry Henderson
+	console.log( oldValue ); //-> Harry Potter
+} );
+
+me.last = "Henderson";
+```
+@codepen
+@highlight 4-8
+
+`getter` properties use [can-observation] internally.  This means that when bound,
+the value of the `getter` is cached and only updates when one of its source
+observables change.  For example:
+
+```js
+import { DefineObject } from "can";
+
+class Person extends DefineObject {
+  static define = {
+    first: String,
+    last: String,
+  };
+
+  get fullName() {
+    console.log( "calculating fullName" );
+    return this.first + " " + this.last;
+  }
+}
+
+const hero = new Person( { first: "Wonder", last: "Woman" } );
+
+console.log( hero.fullName ); // logs Wonder Woman
```

Comment:
These also log `calculating fullName`. I think showing that will make this example more clear.